### PR TITLE
fix: increase CSV escape_max_lines to support long multi-line Golden QA answers

### LINKED ISSUE #5537: [Bug]: Golden QA CSV upload rejects well-formed multi-line answer cells
### Summary

Golden QA CSV uploads with long, multi-paragraph answer cells (e.g. multi-section RTI-style responses with 50+ embedded line breaks) are rejected with "Unable to parse the uploaded CSV file", even when the CSV is well-formed. `CSV.decode/2` is called with `escape_max_lines: 50` — a safety cap on how many lines a quoted field is allowed to span while the parser looks for its closing quote. Legitimate answers longer than 50 lines hit this cap and are misreported as malformed, even though the field is correctly quoted end-to-end.

### Steps to Reproduce

1. Prepare a Golden QA CSV where an answer cell is quoted and spans more than 50 lines (e.g. a multi-section RTI-style response with 50+ embedded line breaks).
2. Upload the CSV via the Golden QA upload flow.
3. Observe the upload is rejected with "Unable to parse the uploaded CSV file", despite the CSV being well-formed and correctly quoted.

### Expected Behaviour

Well-formed CSVs with long multi-line answer cells (up to the existing `@max_golden_qa_file_size` upload cap of 1MB) should parse successfully regardless of how many lines a single quoted field spans.

### Actual Behaviour

Any quoted field spanning more than 50 lines is misreported as malformed due to the `escape_max_lines: 50` cap on `CSV.decode/2`, rejecting otherwise valid uploads.

### Acceptance Criteria

- [ ] `escape_max_lines` is raised from `50` to `1_000_000` (effectively unlimited given the 1MB file size cap, which bounds a file to ~1,048,576 lines even in the degenerate all-newline case).
- [ ] The `escape_max_lines` key is kept explicit in the `CSV.decode/2` call (must not be dropped — the `csv` library's own default is `10`, per `deps/csv/lib/csv/defaults.ex:15`, which would regress this further).
- [ ] A CSV with a multi-line (50+ line) quoted answer cell uploads and parses successfully.
- [ ] Test asserts the multiline answer content reaches `Kaapi.upload_evaluation_dataset/2` unchanged (currently only `golden_qa.name` is asserted in `test/glific_web/resolvers/ai_evaluations_test.exs:763-779` — flagged by CodeRabbit).

### Additional Notes

Fixed in PR #5532 (branch `fix/total_line_allowed_issue_per_row`).

### PR DESCRIPTION
**Pull request by priyanshu6238**
## Summary
- Golden QA CSV uploads with long, multi-paragraph answer cells (e.g. multi-section  RTI-style responses with 50+ embedded line breaks) were being rejected with "Unable to parse the uploaded CSV file", even when the CSV was well-formed.
- Root cause: `CSV.decode/2` was called with `escape_max_lines: 50` — a safety cap on how many lines a quoted field is allowed to span while the parser looks for its closing quote. Legitimate answers longer than 50 lines hit this cap and were misreported as malformed, even though the field was correctly quoted end-to-end.
- Raised the cap to `1_000_000`, which is effectively unlimited given the existing `@max_golden_qa_file_size` upload cap of 1MB (a 1MB file can contain at most ~1,048,576 lines even in the degenerate case of all-newline content).
-  We cannot drop this field entirely. I checked deps/csv/lib/csv/defaults.ex:15: @escape_max_lines 10
    If we drop the escape_max_lines: key from the CSV.decode call entirely, the library falls back to this default of 10.

### CODE REVIEW

File: lib/glific_web/resolvers/ai_evaluations.ex [hunk 1]
```diff
@@ -207,7 +193,7 @@ defmodule GlificWeb.Resolvers.AIEvaluations do
   defp validate_csv_structure(%{path: path}) do
     path
     |> File.stream!()
-    |> CSV.decode(headers: false, escape_max_lines: 50)
+    |> CSV.decode(headers: false, escape_max_lines: 1_000_000)
```
Comment (rvignesh89) [inline lib/glific_web/resolvers/ai_evaluations.ex]: This was looks unreasonably high, what are the side effects of this from a memory on server POV?

Comment (rvignesh89) [inline lib/glific_web/resolvers/ai_evaluations.ex]:
```suggestion
    |> CSV.decode(headers: false, escape_max_lines: 1000)
```
I'd suggest start with a smaller number

**Review by rvignesh89 — CHANGES_REQUESTED** [review]
@max_golden_qa_evaluations 80

@priyanshu6238 can you also remove this validation since kaapi can handle more questions

File: lib/glific_web/resolvers/ai_evaluations.ex (hunk 1 above)
Comment (priyanshu6238) [reply lib/glific_web/resolvers/ai_evaluations.ex]: We already have a file size restriction of 1 MB, so it doesn’t matter . they cannot upload a file larger than 1 MB in total.

Comment (priyanshu6238) [reply lib/glific_web/resolvers/ai_evaluations.ex]: Done

**Comment by priyanshu6238** [comment]
> @max_golden_qa_evaluations 80
> 
> @priyanshu6238 can you also remove this validation since kaapi can handle more questions

Done: https://github.com/glific/glific/pull/5532/changes/f292e326ce6a41b90f4eaeba857d58f18efd69ff

**Comment by priyanshu6238** [comment]
Interesting! While testing different cases, got this error in one of the tests when a single file had more than 100 questions and the duplication factor was set to 1. We still getting this error

Screenshot 2026-08-10 at 11 05 38 PM

**Comment by priyanshu6238** [comment]
> @max_golden_qa_evaluations 80
> 
> @priyanshu6238 can you also remove this validation since kaapi can handle more questions

@rvignesh89 I’ll handle that in another PR. Let’s deploy this one first.

**Comment by shijithkjayan** [comment]
@priyanshu6238 is there a ticket for this?

**Comment by priyanshu6238** [comment]
> @priyanshu6238 is there a ticket for this?

created now: https://github.com/glific/glific/issues/5537

File: test/glific_web/resolvers/ai_evaluations_test.exs [hunk 2]
```diff
@@ -739,6 +739,41 @@ defmodule GlificWeb.Resolvers.AIEvaluationsTest do
       assert msg == "Unable to parse the uploaded CSV file"
     end
 
+    test "accepts a properly quoted answer field spanning more than 50 lines", %{staff: user} do
```
Comment (rvignesh89) [inline test/glific_web/resolvers/ai_evaluations_test.exs]: The test should verify boundary conditions and not arbitrary value in the middle. The previous tests you wrote were more accurate.

**Review by rvignesh89 — APPROVED** [review]

File: test/glific_web/resolvers/ai_evaluations_test.exs (hunk 2 above)
Comment (priyanshu6238) [reply test/glific_web/resolvers/ai_evaluations_test.exs]: Done: https://github.com/glific/glific/pull/5532/changes/670bea42c6ab0f75a4277f90d69d3fbc100a2b91
